controlplane/kubeadm/controllers/controller.go

```diff
@@ -210,6 +210,16 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	}
 
 	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines)
+
+	numMachines := len(ownedMachines)
+	// If the control plane is initialized, wait for health checks to pass to continue.
+	if numMachines > 0 {
+		result, err := r.generalHealthCheck(ctx, cluster, kcp, controlPlane)
+		if err != nil {
+			return result, err
+		}
+	}
```

Comment:
How would this work when we can do remediation? Let's say a Machine isn't responding, if the health check fails, we won't create a new one?